test/unit/stream_alert_rule_processor/test_classifier.py

```diff
@@ -255,46 +410,41 @@ def test_classify_record_kinesis_json_optional(self):
         assert_equal(len(payload.records[0]['key10']), 2)
 
         # record type tests
-        assert_equal(type(payload.records[0]['key1']), list)
-        assert_equal(type(payload.records[0]['key2']), str)
-        assert_equal(type(payload.records[0]['key3']), int)
+        assert_is_instance(payload.records[0]['key1'], list)
+        assert_is_instance(payload.records[0]['key2'], str)
+        assert_is_instance(payload.records[0]['key3'], int)
 
-    def test_classify_record_kinesis_json(self):
-        """Payload Classify JSON - boolean, float, integer types"""
+    def test_json_type_casting(self):
+        """StreamClassifier - JSON with various types (boolean, float, integer)"""
         kinesis_data = json.dumps({
             'key4': 'true',
             'key5': '10.001',
             'key6': '10',
             'key7': False
         })
-        payload = self.payload_generator(kinesis_stream='test_kinesis_stream',
-                                         kinesis_data=kinesis_data)
-        classifier = StreamClassifier(config=self.config)
-        classifier.map_source(payload)
 
-        # pre parse and classify
-        data = self.pre_parse_kinesis(payload)
-        classifier.classify_record(payload, data)
+        service, entity = 'kinesis', 'test_kinesis_stream'
+        raw_record = _make_kinesis_raw_record(entity, kinesis_data)
+        payload = self._prepare_and_classify_payload(service, entity, raw_record)
 
         # valid record test
         assert_equal(payload.valid, True)
-        assert_equal(type(payload.records[0]), dict)
+        assert_is_instance(payload.records[0], dict)
```

Comment:
It is OK to check the type directly if you want; `isinstance` is a looser check because it will work for subclasses as well

```python
d = collections.OrderedDict()
type(d) == dict  # False
isinstance(d, dict)  # True
```